core/src/main/scala/kafka/message/ByteBufferMessageSet.scala

```diff
@@ -111,6 +181,20 @@ object ByteBufferMessageSet {
   }
 }
 
+private class OffsetAssigner(offsets: Seq[Long]) {
+  val index = new AtomicInteger(0)
+
+  def this(offsetCounter: AtomicLong, size: Int) {
+    this((offsetCounter.get() to offsetCounter.get + size).toSeq)
+    offsetCounter.addAndGet(size)
+  }
+
+  def nextAbsoluteOffset = offsets(index.getAndIncrement)
+
+  def toRelativeOffset(offset: Long) = offset - offsets(0)
```

Comment:
The confusion is that toRelativeOffset() doesn't really return the relative offset as defined in line 150. Would it be better to rename it to toInnerOffset()?